### PR TITLE
#19 - Added package metadata route

### PR DESCRIPTION
**Pull request by olegmoz**
Part of issue #19
Added package metadata service route

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#55](https://github.com/artipie/nuget-adapter/issues/55) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/c1e42489-afae-427a-acdf-ae1d7602b242); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artpie/nuget/http/metadata/NuGetPackageMetadataTest.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        final Response response = this.nuget.response(
+            "PUT /base/registrations/newtonsoft.json/index.json",
+            Collections.emptyList(),
+            Flowable.empty()
+        );
+        MatcherAssert.assertThat(response, new RsHasStatus(RsStatus.METHOD_NOT_ALLOWED));
+    }
+
+    /**
+     * Matcher for bytes array representing valid Service Index JSON.
```
Comment (Vatavuk) [inline src/test/java/com/artpie/nuget/http/metadata/NuGetPackageMetadataTest.java]: @olegmoz It's not service index JSON.

File: src/test/java/com/artpie/nuget/http/metadata/NuGetPackageMetadataTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+     *
+     * @since 0.1
+     */
+    private static class IsValidPages extends TypeSafeMatcher<byte[]> {
+
+        @Override
+        public void describeTo(final Description description) {
+            description.appendText("is pages JSON");
```
Comment (Vatavuk) [inline src/test/java/com/artpie/nuget/http/metadata/NuGetPackageMetadataTest.java]: @olegmoz naming convention here as well.

**Comment by olegmoz** [comment]
@Vatavuk `Registration` name makes more sense, thank you! Made the changes, please review

**Comment by g4s8** [comment]
@olegmoz please resolve merge conflicts

**Comment by olegmoz** [comment]
@g4s8 conflicts resolved

**Comment by 0crat** [comment]
Job was finished in 22 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Pull request merged**